From 26e70ee8f7d8fb1b80e9c14ccfbef240a66e84dc Mon Sep 17 00:00:00 2001
From: Behan Webster <behanw@converseincode.com>
Date: Tue, 3 Sep 2013 22:27:26 -0400
Subject: [PATCH] Kbuild, arm64, LLVMLinux: Add Kbuild support for building
 arch arm64 with Clang

When compiling with clang, turn off builtins, and warnings which are too noisy
when compiling the arm64 kernel.

Author: Behan Webster <behanw@converseincode.com>
Signed-off-by: Behan Webster <behanw@converseincode.com>
Reviewed-by: Mark Charlebois <charlebm@gmail.com>
---
 arch/arm64/Makefile | 6 ++++++
 1 file changed, 6 insertions(+)

diff --git a/arch/arm64/Makefile b/arch/arm64/Makefile
index 2df5e5d..80d1153 100644
--- a/arch/arm64/Makefile
+++ b/arch/arm64/Makefile
@@ -19,6 +19,12 @@ LIBGCC 		:= $(shell $(CC) $(KBUILD_CFLAGS) -print-libgcc-file-name)
 
 KBUILD_DEFCONFIG := defconfig
 
+ifeq ($(COMPILER),clang)
+KBUILD_CFLAGS   += -fno-builtin
+KBUILD_CFLAGS   += -Wno-asm-operand-widths
+KBUILD_CFLAGS   += -Wno-initializer-overrides
+endif
+
 KBUILD_CFLAGS	+= -mgeneral-regs-only
 ifeq ($(CONFIG_CPU_BIG_ENDIAN), y)
 KBUILD_CPPFLAGS	+= -mbig-endian
-- 
1.9.1

